File: tests/fixtures/xpath_injection/python/unsafe_xpath_query.py
# Unsafe: tainted form data concatenated into an XPath expression and passed
# to lxml's `tree.xpath()`.  Suffix matching on `xpath` catches the
# bound-receiver call directly.
from lxml import etree
from flask import request


def lookup():
    tree = etree.parse("users.xml")
    user = request.form["user"]
    expr = "//user[name='" + user + "']"
    return tree.xpath(expr)

File: tests/fixtures/xpath_injection/ruby/unsafe_xpath_query.rb
# Unsafe: Sinatra params concatenated into an XPath expression and passed to
# Nokogiri's `xpath` method.  Suffix matching on `xpath` catches the
# bound-receiver call directly.
require 'nokogiri'

def lookup(params)
  doc = Nokogiri::XML(File.read("users.xml"))
  user = params[:user]
  expr = "//user[name='#{user}']"
  doc.xpath(expr)
end
